packages/adapters/cache/src/lib/caches/auctions.ts

```diff
@@ -131,7 +131,11 @@ export class AuctionsCache extends Cache {
     let keys: string[] = [];
     await new Promise((res) => {
       stream.on("data", (resultKeys: string[] = []) => {
-        keys = keys.concat(resultKeys);
+        // Note that resultKeys will sometimes contain duplicates due to SCAN's implementation in Redis
+        // link : https://redis.io/commands/scan/#scan-guarantees
+        for (const resultKey of resultKeys) {
```

Comment:
We need to dedup the bids inserting INTO the cache, not reading from the cache.

Reply:
we don't add the same bid more than once.  the problem exists when reading